src/stacks-hierarchy/Root_PurchaseHarborSearchScreen/Root_PurchaseHarborSearchScreen.tsx

```diff
@@ -0,0 +1,200 @@
+import {FullScreenHeader} from '@atb/components/screen-header';
+import {StyleSheet} from '@atb/theme';
+import {useTranslation} from '@atb/translations';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {
+  ActivityIndicator,
+  Keyboard,
+  TextInput as InternalTextInput,
+  View,
+} from 'react-native';
+import {RootStackScreenProps} from '@atb/stacks-hierarchy/navigation-types';
+import {TextInputSectionItem} from '@atb/components/sections';
+import {MessageBox} from '@atb/components/message-box';
+import {ScrollView} from 'react-native-gesture-handler';
+import {useIsFocused} from '@react-navigation/native';
+import {useDebounce} from '@atb/utils/useDebounce';
+import {HarborResults} from '@atb/stacks-hierarchy/Root_PurchaseHarborSearchScreen/HarborResults';
+import {useIsLoading} from '@atb/utils/use-is-loading';
+import {ErrorType} from '@atb/api/utils';
+import {ScreenReaderAnnouncement} from '@atb/components/screen-reader-announcement';
+import {
+  TransportMode,
+  TransportSubmode,
+} from '@atb/api/types/generated/journey_planner_v3_types';
+import {
+  getStopPlaceConnections,
+  getStopPlacesByMode,
+} from '@atb/api/stop-places';
+import {StopPlace, StopPlaces} from '@atb/api/types/stopPlaces';
+import HarborSearchTexts from '@atb/translations/screens/subscreens/HarborSearch';
+import {translateErrorType} from '@atb/stacks-hierarchy/utils';
+
+type Props = RootStackScreenProps<'Root_PurchaseHarborSearchScreen'>;
+
+export const Root_PurchaseHarborSearchScreen = ({navigation, route}: Props) => {
+  const {fromHarbor, fareProductTypeConfig, preassignedFareProduct} =
+    route.params;
+
+  const {t} = useTranslation();
+  const inputRef = useRef<InternalTextInput>(null);
+  const [text, setText] = useState('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  const onSave = (selectedStopPlace: StopPlace) => {
+    selectedStopPlace &&
+      navigation.navigate({
+        name: 'Root_PurchaseOverviewScreen',
+        params: {
+          mode: 'Ticket',
+          fareProductTypeConfig,
+          preassignedFareProduct,
+          fromHarbor: fromHarbor ?? selectedStopPlace,
+          toHarbor: fromHarbor ? selectedStopPlace : undefined,
+        },
+        merge: true,
+      });
+  };
+
+  const styles = useStyles();
+  // capturing focus on mount and on press
+  const focusInput = () => setTimeout(() => inputRef.current?.focus(), 0);
+  const isFocused = useIsFocused();
+
+  useEffect(() => {
+    if (isFocused) focusInput();
```

Comment:
Rather inline the `focusInput` function. By creating it during render it becomes a [reactive value](https://react.dev/learn/lifecycle-of-reactive-effects#effects-react-to-reactive-values), and should be in the dependency list.

> Props, state, and other values declared inside the component are reactive because they’re calculated during rendering and participate in the React data flow. (...) Reactive values must be included in dependencies